runtime/oti/VMHelpers.hpp

```diff
@@ -2077,71 +2109,142 @@ class VM_VMHelpers
 	isConcurrentlyScannedFromContinuationState(uintptr_t continuationState, bool isGlobalGC)
 	{
 		uintptr_t concurrentGCMask = getConcurrentGCMask(isGlobalGC);
-		return J9_ARE_ANY_BITS_SET(continuationState, concurrentGCMask);
+		return J9_ARE_ALL_BITS_SET(continuationState, concurrentGCMask);
+	}
+
+	static VMINLINE uintptr_t
+	setConcurrentlyScannedToContinuationState(uintptr_t continuationState, bool isGlobalGC)
+	{
+		return continuationState | getConcurrentGCMask(isGlobalGC);
+	}
+
+	static VMINLINE uintptr_t
+	resetConcurrentlyScannedToContinuationState(uintptr_t continuationState, bool isGlobalGC)
```

Comment:
These "reset" functions are poorly named. They don't in fact reset anything, they're just getters.

Reply:
update function names from set/reset to  getNewContinuationStateWith... 
 setCarrierAndPendingStateToContinuationState  ---->  getNewContinuationStateWithSettingCarrierAndPendingState
 resetPendingToContinuationState --->  getNewContinuationStateWithResettingPendingState
 resetConcurrentlyScannedToContinuationState  --->  getNewContinuationStateWithResettingConcurrentlyScanned
 setConcurrentlyScannedToContinuationState  ---> getNewContinuationStateWithSettingConcurrentlyScanned

Reply:
there are a group of methods
- that deal with a local instance of state and they have To/FromContinuationState in their names; the state parameter is passed by value, and new value (set or reset) is returned via C 'return'
- that deal with the state within a passes continuation, and continuation is passes by address, so they can directly mutate the state without using the return   


How about this?
1) make the first group pass the state by address too
Instead of 
```uintptr_t newContinuationState = resetConcurrentlyScannedToContinuationState(oldContinuationState)```
we would have 
```
uintptr_t newContinuationState = oldContinuationState;
resetConcurrentlyScannedToContinuationState(&newContinuationState);
```
2)  introduce
```typedef ContinuationState uintptr_t;```
and simplify the name of the methods in the first group by removing To/FromContinuationState 
For example 
```uintptr_t	resetConcurrentlyScannedToContinuationState(uintptr_t continuationState);```
 would become
``` void  	resetConcurrentlyScanned(ContinuationState *continuationState);```

Or any other specific suggestion from your side?

Reply:
> update function names from set/reset to getNewContinuationStateWith...
> 
> ```
> setCarrierAndPendingStateToContinuationState  ---->  getNewContinuationStateWithSettingCarrierAndPendingState
> resetPendingToContinuationState --->  getNewContinuationStateWithResettingPendingState
> resetConcurrentlyScannedToContinuationState  --->  getNewContinuationStateWithResettingConcurrentlyScanned
> setConcurrentlyScannedToContinuationState  ---> getNewContinuationStateWithSettingConcurrentlyScanned
> ```

if we go with this, I believe it's more appropriate to say 'by' than 'with'

Reply:
and no need to need to say 'New',

Reply:
Definitely looks better, though I would remove the "from continuation state" from the names (since the only parameter is the continuation state, it seems redundant).

Reply:
Ah, I see Aleks already mentioned this.

Reply:
remove "FromContinuationState" from helper functions

getNewContinuationStateWithSettingCarrierAndPendingState  --> void settingCarrierAndPendingState(ContinuationState *continuationState, J9VMThread *carrierThread)
getNewContinuationStateWithResettingPendingState --> void resetPendingState(ContinuationState *continuationState)
getNewContinuationStateWithResettingConcurrentlyScanned --> void resetConcurrentlyScanned(ContinuationState *continuationState, bool isGlobalGC)
getNewContinuationStateWithSettingConcurrentlyScanned  --> void setConcurrentlyScanned(ContinuationState *continuationState, bool isGlobalGC)